### ISSUE TITLE
UIAccessibility issue with scrolling

### ISSUE THREAD

**Issue by palaniraja** [nicklockwood/iCarousel#346]
It seems not compatible with UIAccessibility. For eg., scrollview with voiceover is accessible with three finger scrolling,  but iCarousel is missing the feature.

**Comment by dral3x** [comment]
3 and half year has passed. No new on this issue?

**Comment by acecilia** [comment]
Seems like there are no answers about this topic. I also need the accessibility features. There are already two open pull requests claiming to add the support. @nicklockwood can you give us some feedback? Thanks

**Comment by dral3x** [comment]
We implemented accessibility in the simplest way possible. Take a look at our fork if you want https://github.com/spreaker/iCarousel

**Comment by kaannesinn** [comment]
If you comment the lines below on iCarousel.m file, iCarousel view and its custom subviews will be accessible.

//set up accessibility
// self.accessibilityTraits = UIAccessibilityTraitAllowsDirectInteraction;
// self.isAccessibilityElement = YES;

**Comment by acecilia** [comment]
What I end up doing is customizing the value of `accessibilityLabel` and `accessibilityValue` in the subview I put in the carousel items:

```
    private func accessibilitySetup() {
        accessibilityLabel = "Carousel item"
        let messageElements = [
            bgCircleTitleLabel.text,
            bgCircleSubtitleLabel.text,
            bottomTitleLabel.text,
            bottomSubtitleLabel.text
        ].compactMap { $0?.isEmpty == false ? $0 : nil }
        accessibilityValue = messageElements.joined(separator: " ")
    }
```
The elements inside the `messageElements` array are subviews of the carousel item. That function just call it when setting up the carousel item (everytime the item is changed).
